tools/jdk/java_tools_defs.bzl

```diff
@@ -0,0 +1,91 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Bazel rules for creating Java toolchains embedded into java_tools."""
+
+DEFAULT_JVM_OPTS = [
+   # In JDK9 we have seen a ~30% slow down in JavaBuilder performance when using
+   # G1 collector and having compact strings enabled.
+   "-XX:+UseParallelOldGC",
+   "-XX:-CompactStrings",
+   # Allow JavaBuilder to access internal javac APIs.
+   "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+   "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+   "--add-opens=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+
+   # override the javac in the JDK.
+   "--patch-module=java.compiler=$(location :java_compiler_jar)",
+   "--patch-module=jdk.compiler=$(location :jdk_compiler_jar)",
```

Comment:
I'm actually confused here, so I'll ask :) Is it true that this relative label keeps referencing `@bazel_tools//tools/jdk:java_compiler_jar` even when we call the macro from a random package `//foo/bar`? Or it assumed that the //foo/bar package has those labels around?